bedwars-plugin/src/main/java/com/andrei1058/bedwars/listeners/GameEnd.java

```diff
@@ -0,0 +1,41 @@
+package com.andrei1058.bedwars.listeners;
+
+import com.andrei1058.bedwars.api.arena.generator.IGenerator;
+import com.andrei1058.bedwars.api.events.gameplay.GameEndEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Item;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.UUID;
+
+public class GameEnd implements Listener {
+
+    @EventHandler
+    public void stopPickUP(GameEndEvent event){
+        if (!event.getAliveWinners().isEmpty()){
+            for (UUID p : event.getAliveWinners()) {
+                Bukkit.getPlayer(p).setCanPickupItems(false);
+                Bukkit.getPlayer(p).getInventory().clear();
+            }
+                for (IGenerator gen : event.getArena().getOreGenerators()) {
```

Comment:
we can simply do a check on arena state in generator#spawn

Reply:
I have addressed this